packages/malloy-vscode/src/common/connection_manager.ts

```diff
@@ -18,149 +18,185 @@ import { DuckDBConnection } from "@malloydata/db-duckdb";
-            password = connectionConfig.password;
-          } else if (connectionConfig.useKeychainPassword) {
-            password =
-              (await getPassword(
-                "com.malloy-lang.vscode-extension",
-                `connections.${connectionConfig.id}.password`
-              )) || undefined;
-          }
-          return {
-            username: connectionConfig.username,
-            host: connectionConfig.host,
-            password,
-            port: connectionConfig.port,
-            databaseName: connectionConfig.databaseName,
-          };
-        };
-        return new PostgresConnection(
-          connectionConfig.name,
-          () => ({ rowLimit: this.getCurrentRowLimit() }),
-          configReader
-        );
-      }
-      case ConnectionBackend.DuckDB: {
-        if (!isDuckDBAvailable) {
-          throw new Error("DuckDB is not available.");
-        }
-        try {
-          return new DuckDBConnection(
-            connectionConfig.name,
-            ":memory:",
-            connectionConfig.workingDirectory
-          );
-        } catch (error) {
-          console.log("Could not create DuckDB connection:", error);
-          throw error;
+    return getConnectionForConfig(connectionConfig);
+  }
+
+  public getConnectionLookup(url: URL): LookupConnection<Connection> {
+    const workingDirectory = path.dirname(url.pathname);
+    if (!this.connectionLookups[workingDirectory]) {
+      this.connectionLookups[workingDirectory] = new DynamicConnectionLookup(
+        this.configs,
+        {
+          workingDirectory,
+          rowLimit: this.getCurrentRowLimit(),
         }
-      }
+      );
     }
+    return this.connectionLookups[workingDirectory];
   }
 
-  getConnectionManager(url: URL): LookupConnection<Connection> {
-    const dirname = path.dirname(url.pathname);
-    if (!this._managers[dirname]) {
-      this._managers[dirname] = new DynamicConnectionManager(
-        this.connections,
-        dirname
-      );
+  public getCurrentRowLimit(): number | undefined {
+    return undefined;
+  }
+
+  protected static filterUnavailableConnectionBackends(
+    connectionsConfig: ConnectionConfig[]
+  ): ConnectionConfig[] {
+    return connectionsConfig.filter(
+      (config) =>
+        isDuckDBAvailable || config.backend !== ConnectionBackend.DuckDB
+    );
+  }
+
+  buildConfigMap(configs: ConnectionConfig[]): void {
+    if (configs.length === 0) {
+      configs = [
+        {
+          name: "bigquery",
+          backend: ConnectionBackend.BigQuery,
+          id: "bigquery-default",
+          isDefault: true,
+        },
+        {
+          name: "duckdb",
+          backend: ConnectionBackend.DuckDB,
+          id: "duckdb-default",
+          isDefault: false,
+        },
+      ];
     }
-    return this._managers[dirname];
+    configs.forEach((config) => {
+      if (config.isDefault) {
+        this.configs[DEFAULT_CONFIG] = config;
```

Comment:
Maybe we should do a `this.defaultConfig`  or at least use a `Symbol("default-config")` just so we don't prevent someone from naming their connection "__default__" (not sure why you would do that, but someone might want to!)